### PR TITLE
fix(rln-relay): feature guard

### PR DESCRIPTION
**Pull request by rymnc**
Fixes the missing `when defined(rln)` feature guard for builds which do not require RLN - 
https://github.com/status-im/nwaku/issues/1372, also does not add `RateLimitProof` to the WakuMessage payload if rln is not compiled.

### CODE REVIEW

**Comment by LNSD** [comment]
I understand the "feature guard" is necessary when using the types exported by the RLN module, e.g., in the `wakunode2` code, the `waku_node` code, etc.

But in the RLN protocol modules, it should not be necessary. The imports of those modules are the ones that should be conditional imported at compile time.

**Review by LNSD — APPROVED** [review]
LGTM ✅

The next thing I find necessary is putting all modules behind an "entry point" module as other protocols have.

File: waku/v2/protocol/waku_message.nim [hunk 1]
```diff
@@ -40,7 +43,8 @@ type WakuMessage* = object
     # the proof field indicates that the message is not a spam
```
[the remark this answers is not in the document]
Comment (rymnc) [reply waku/v2/protocol/waku_message.nim:43]: Will resolve in a subsequent PR, macOS CI is taking forever 😔

**Comment by rymnc** [comment]
Merging the PR since the failing test refers to #1368

**Pull request merged**